eos/utils/omnes-factor-impl.hh

```diff
@@ -0,0 +1,307 @@
+
+        if (_b)
+            gsl_vector_free(_b);
+        _b = nullptr;
+    }
+
+    // Prepare system of equations
+    template <unsigned order_, unsigned nints_>
+    void OmnesFactor<order_, nints_>::PrepSys(const double & bcpos)
+    {
+        std::array<double, (nints_ + 1)*order_> slist;
+        std::array<double, (nints_ + 1)*order_> bc;
+
+        for (unsigned i = 0; i < nints_ - 1; i++)
+        {
+            std::array<double, order_> bcslice = RRab(bcpos, _intervals[i], _intervals[i + 1]);
+            for (unsigned j = 0; j < order_; j++)
+            {
+                slist[i*order_ + j] = (_intervals[i] + _intervals[i + 1] + (_intervals[i + 1] - _intervals[i])*_zeros[j])/2;
+                bc[i*order_ + j] = bcslice[j];
+            }
+        }
+        std::array<double, order_> bcend = RRinf(bcpos, _intervals[nints_ - 1]);
+        for (unsigned j = 0; j < order_; j++)
+        {
+            slist[(nints_ - 1)*order_ + j] = (2*_intervals[nints_ - 1]/(1 - _zeros[j]));
+            bc[(nints_ - 1)*order_ + j] = bcend[j];
+        }
+
+        for (unsigned i = 0; i < nints_*order_; i++)
+        {
+            for (unsigned j = 0; j < nints_; j++)
+            {
+                std::array<double, order_> row;
+                if (j != nints_ - 1)
+                {
+                    row = RRab(slist[i], _intervals[j], _intervals[j + 1]);
+                }
+                else
+                {
+                    row = RRinf(slist[i], _intervals[j]);
+                }
+                for (unsigned k = 0; k < order_; k++)
+                {
+                    if (i == j*order_ + k)
+                    {
+                        gsl_matrix_set(_sys, i, j*order_ + k, 1. - row[k]);
+                    }
+                    else
+                    {
+                        gsl_matrix_set(_sys, i, j*order_ + k, -row[k]);
+                    }
+                }
+            }
+        }
+
+        for (unsigned j = 0; j < nints_*order_; j++)
+        {
+            gsl_matrix_set(_sys, nints_*order_, j, bc[j]);
+        }
+    }
+
+    // Solve the system
+    template <unsigned order_, unsigned nints_>
+    void OmnesFactor<order_, nints_>::SolveSys()
+    {
+        // Initialize b & sys2
+        gsl_vector_set(_b, nints_*order_, 1.);
+        gsl_matrix_memcpy(_sys2, _sys);
+
+        // Solve things, TODO: Check for error output....
+        gsl_linalg_SV_decomp(_sys2, _V, _S, _work);
+        gsl_linalg_SV_solve(_sys2, _V, _S, _b, _x);
+
+        // Convert gsl -> vector
+        for (unsigned i = 0; i < nints_*order_; i++)
+        {
+            _sol[i] = gsl_vector_get(_x, i);
+        }
+
+        _solved = true;
+        gsl_blas_dgemv(CblasNoTrans, 1., _sys, _x, -1., _b);
+        _err = gsl_blas_dnrm2(_b);
+    }
+
+    // Compute sums of Eq. 58 in [M:1999A]
+    template <unsigned order_, unsigned nints_>
+    std::array<double, order_> OmnesFactor<order_, nints_>::LQSum(const double & z)
```

Comment:
(coding style) This member function is not a named constructor, correct? In that case, its name should be ``l_q_sum`` or similar.